lib/pf/Switch/Juniper/EX2300.pm

```diff
@@ -116,8 +116,17 @@ sub setAdminStatus {
     my $logger = $self->logger;
 
     #We need to fetch the MAC on the ifIndex in order to bounce switch port with CoA.
-    my @locationlog = locationlog_view_open_switchport( $self->{_ip}, $ifIndex );
+    my @locationlog = locationlog_view_open_switchport_no_VoIP( $self->{_ip}, $ifIndex );
     my $mac = $locationlog[0]->{'mac'};
+    if (!$mac) {
+        @locationlog = locationlog_view_open_switchport_only_VoIP( $self->{_ip}, $ifIndex );
+        $mac = $locationlog[0]->{'mac'};
+    }
+    
+    if (!$mac) {
+        $logger->info("Can't find MAC address in the locationlog... we won't perform port bounce");
+        return 1;
```

Comment:
Add "use pf::constants" at the top of the file and use '$TRUE' instead of the hardcoded '1'